### PR TITLE
Hide slider when image dimension is 1

### LINKED ISSUE #421: Slider should disappear when only 1 slice is visualised
When only 1 slice is visualised, the slider shows values between -1 and 0. The slider should not appear and only one slice should be shown.

This is connected to #420

### PR DESCRIPTION
**Pull request by DanicaSTFC**
closes #421

### CODE REVIEW

File: Wrappers/Python/ccpi/viewer/CILViewer2D.py [hunk 1]
```diff
@@ -1586,6 +1592,9 @@ def installSliceSliderWidgetPipeline(self):
         self.sliderWidget = sw
         self.sliderCallback = cb
 
+        # enable slider
+        self.sliderWidget.SetEnabled(not slider_hidden)
```
Comment (paskino) [inline Wrappers/Python/ccpi/viewer/CILViewer2D.py:1596]: Why not put this at line 1576?

**Comment by paskino** [comment]
From email from @DanicaSTFC 
> I know you have a flag for this already but I preferr to keep the hide/show flag separate from the creation flag.

At the time I did not find a way to hide the slider widget, which you seem to have found. I used `EnabledOn`.

https://github.com/TomographicImaging/CILViewer/blob/cf6cb86489d1962caa8e2195d9499c3ffbc6e2a7/Wrappers/Python/ccpi/viewer/CILViewer2D.py#L1565-L1569

It would be useful to add a method `setSliderWidgetEnabled` to be able to control the visibility of the slider widget.

File: Wrappers/Python/ccpi/viewer/CILViewer2D.py (hunk 1 above)
Comment (DanicaSTFC) [reply Wrappers/Python/ccpi/viewer/CILViewer2D.py:1596]: I moved it

**Comment by DanicaSTFC** [comment]
> This PR seems to fix the last point at [#365 (comment)](https://github.com/TomographicImaging/CILViewer/pull/365#issuecomment-1737227437)
> 
> From email from @DanicaSTFC
> 
> > I know you have a flag for this already but I preferr to keep the hide/show flag separate from the creation flag.
> 
> At the time I did not find a way to hide the slider widget, which you seem to have found. I used `EnabledOn`.
> 
> https://github.com/TomographicImaging/CILViewer/blob/cf6cb86489d1962caa8e2195d9499c3ffbc6e2a7/Wrappers/Python/ccpi/viewer/CILViewer2D.py#L1565-L1569
> 
> It would be useful to add a method `setSliderWidgetEnabled` to be able to control the visibility of the slider widget. This would allow a proper treatment of the visibility and creation of the widget.

I did not create a new method to control the visibility as the `SetEnabled` method already exists. The creation of the widget in a viewer is a different matter. I was not sure what you wanted to do with the enabled parameter. Perhaps we can discuss this in a meeting.

**Pull request merged**